internal/serve/httphandler/forgot_password_handler_test.go

```diff
@@ -57,51 +61,82 @@ func Test_ForgotPasswordHandler(t *testing.T) {
 		SDPUIBaseURL: &uiBaseURL,
 	}
 	ctx := tenant.SaveTenantInContext(context.Background(), &tnt)
-
-	t.Run("Should return http status 200 on a valid request", func(t *testing.T) {
-		requestBody := `
-		{ 
-			"email": "valid@email.com" ,
-			"recaptcha_token": "validToken"
-		}`
-
-		rr := httptest.NewRecorder()
-		req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, strings.NewReader(requestBody))
-		require.NoError(t, err)
-
+	const email = "valid@email.com"
+	defaultUserRoles := []string{data.OwnerUserRole.String()}
+	user := &auth.User{
+		ID:    "userID",
+		Email: email,
+	}
+	const reCAPTCHAToken = "validToken"
+	defaultReqBody := fmt.Sprintf(`
+				{ 
+					"email": "%s",
+					"recaptcha_token": "%s"
+				}`, user.Email, reCAPTCHAToken)
+
+	userByEmailSetup := func(user *auth.User, userRoles []string) {
 		authenticatorMock.
-			On("ForgotPassword", req.Context(), "valid@email.com").
-			Return("resetToken", nil).
+			On("GetUserByEmail", mock.Anything, user.Email).
+			Return(user, nil).
 			Once()
-		reCAPTCHAValidatorMock.
-			On("IsTokenValid", mock.Anything, "validToken").
-			Return(true, nil).
+		roleManagerMock.
+			On("GetUserRoles", mock.Anything, user).
+			Return(userRoles, nil).
 			Once()
+	}
+	t.Run("Should return http status 200 on a valid request", func(t *testing.T) {
+		usersRoles := make([][]string, 2)
+		// role cannot bypass reCAPTCHA
+		usersRoles[0] = []string{data.OwnerUserRole.String()}
+		// role can bypass reCAPTCHA
+		usersRoles[1] = []string{data.OwnerUserRole.String(), data.APIUserRole.String()}
+		for _, userRoles := range usersRoles {
+			requestBody := fmt.Sprintf(`
+				{ 
+					"email": "%s"
+				}`, user.Email)
```

Comment:
💡  This setup is confusing to me. Can you create two users, one with the API role and one that doesn't have it? That would make things easier to read and also make the code easier to maintain.

Reply:
I fixed this to include a user with each of the list of user roles! let me know if it's still unclear to you